importer/indico_importer/controllers.py

```diff
@@ -36,3 +44,72 @@ def _process(self):
         with plugin.plugin_context():
             data = {'records': importer.import_data(query, size)}
         return jsonify(data)
+
+
+class RHEndTimeBase(RHManageTimetableBase):
+    """Base class for the importer operations"""
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.event_new
+        },
+        'preserved_args': {
+            'importer_name'
+        }
+    }
+
+    @staticmethod
+    def _find_latest_end_dt(entries):
+        latest_dt = None
+        for entry in entries:
+            if latest_dt is None or entry.end_dt > latest_dt:
+                latest_dt = entry.end_dt
+        return latest_dt
+
+    def _format_date(self, date):
+        return date.astimezone(timezone(self.event_new.timezone)).strftime('%H:%M')
+
+
+class RHDayEndTime(RHEndTimeBase):
+    """Get the end_dt of the latest timetable entry or the event start_dt if no entry exist on that date"""
+
+    def _checkParams(self, params):
+        RHEndTimeBase._checkParams(self, params)
+        self.date = self.event_new.tzinfo.localize(datetime.strptime(request.args['selectedDay'], '%Y/%m/%d')).date()
+
+    def _process(self):
+        end_date = self.date + relativedelta(days=1)
+        print self.date
+        event_start_date = db.cast(TimetableEntry.start_dt.astimezone(self.event_new.tzinfo), db.Date)
+        entries = self.event_new.timetable_entries.filter(event_start_date == self.date)
+        latest_end_dt = self._find_latest_end_dt(entries)
+        if latest_end_dt is None:
+            event_start = self.event_new.start_dt.astimezone(self.date.tzinfo)
+            latest_end_dt = self.date.replace(hour=event_start.hour, minute=event_start.minute)
```

Comment:
ah no that fails in the next line. `self.date` needs to stay a datetime (and just call `.date()` for the usage in the query)